src/client/signatures/rc_crypto_verifier.rs

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+use {
+    super::x509::fetch_public_key,
+    super::{Collection, SignatureError, Verification},
+    canonical_json,
+    log::debug,
+    rc_crypto::signature,
+    serde_json::json,
+};
+
+pub struct RcCryptoVerifier {}
+
+impl RcCryptoVerifier {}
+
+impl Verification for RcCryptoVerifier {
+    fn verify(&self, collection: &Collection) -> Result<(), SignatureError> {
```

Comment:
This looks mostly identical to the ring implementation, so there seems to be an opportunity to factor out the common code, e.g. into a default implementation on the `Verification` trait.